pkg/virt-api/api.go

```diff
@@ -494,33 +493,39 @@ func (app *virtAPIApp) createSubresourceApiservice() error {
 		}
 	}
 
-	if registerApiService {
-		_, err = aggregatorClient.ApiregistrationV1beta1().APIServices().Create(&apiregistrationv1beta1.APIService{
-			ObjectMeta: metav1.ObjectMeta{
-				Name:      subresourceAggregatedApiName,
-				Namespace: namespace,
-				Labels: map[string]string{
-					v1.AppLabel: "virt-api-aggregator",
-				},
+	newApiService := &apiregistrationv1beta1.APIService{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      subresourceAggregatedApiName,
+			Namespace: namespace,
+			Labels: map[string]string{
+				v1.AppLabel: "virt-api-aggregator",
 			},
-			Spec: apiregistrationv1beta1.APIServiceSpec{
-				Service: &apiregistrationv1beta1.ServiceReference{
-					Namespace: namespace,
-					Name:      "virt-api",
-				},
-				Group:                v1.SubresourceGroupName,
-				Version:              v1.SubresourceGroupVersion.Version,
-				CABundle:             app.signingCertBytes,
-				GroupPriorityMinimum: 1000,
-				VersionPriority:      15,
+		},
+		Spec: apiregistrationv1beta1.APIServiceSpec{
+			Service: &apiregistrationv1beta1.ServiceReference{
+				Namespace: namespace,
+				Name:      "virt-api",
 			},
-		})
+			Group:                v1.SubresourceGroupName,
+			Version:              v1.SubresourceGroupVersion.Version,
+			CABundle:             app.signingCertBytes,
+			GroupPriorityMinimum: 1000,
+			VersionPriority:      15,
+		},
+	}
+
+	if registerApiService {
+		_, err = aggregatorClient.ApiregistrationV1beta1().APIServices().Create(newApiService)
 		if err != nil {
 			return err
 		}
-	} else if reflect.DeepEqual(apiService.Spec.CABundle, app.signingCertBytes) == false {
-		// Update apiService if CA bundle doesn't match ours
-		apiService.Spec.CABundle = app.signingCertBytes
+	} else {
+		if apiService.ObjectMeta.Namespace != namespace {
```

Comment:
Ha, now I see the issue. That should be:

```golang
if apiService.Spec.Service.Namespace  != namespace {

}

Reply:
whoa, hooray for functional tests :)